nncf/torch/tensor_statistics/algo.py

```diff
@@ -41,6 +43,28 @@ def __eq__(self, other: "TensorStatisticObservationPoint"):
         return self.target_point == other.target_point
 
 
+def register_inputs_hook_factory(collector: TensorCollector) -> Callable[[torch.Tensor], torch.Tensor]:
```

Comment:
A [factory pattern](https://en.wikipedia.org/wiki/Factory_method_pattern) usually means that the return value will have more than one possibility for its derived type, which all inherit from a same interface. You are only ever going to instantiate a specific callable here. I would recommend renaming it to `create_register_input_hook` if you get the chance

Reply:
Done